jadx-core/src/main/java/jadx/core/xmlgen/ResTableParser.java

```diff
@@ -315,7 +316,14 @@ private String getResName(int resRef, String origKeyName) {
 			constField.add(AFlag.DONT_RENAME);
 			return constField.getName();
 		}
-		return "RES_" + resRef; // autogenerate key name
+		// Making sure origKeyName compliant with resource file name rules
+		Matcher m = VALID_RES_KEY_PATTERN.matcher(origKeyName);
+		StringBuilder sb = new StringBuilder();
+		while (m.find()) {
+			sb.append(m.group());
+		}
+		// autogenerate key name, appended with converted origKeyName to be human-friendly
+		return ("res_" + resRef + "_" + sb.toString()).toLowerCase();
```

Comment:
Effectively you are cleaning out all disallowed characters from the resource name. In my opinion it is better for readability to have a separator between the different cleaned sections. 

Also keep in mind that on heavily obfuscated apps the cleaned resource name may be empty as all characters were invalid. Therefore I would propose to generate the new resource name this way:

		// Making sure origKeyName compliant with resource file name rules
		Matcher m = VALID_RES_KEY_PATTERN.matcher(origKeyName);
		StringBuilder sb = new StringBuilder();
		boolean first = true;
		while (m.find()) {
			if (!first) {
				sb.append("_");
			}
			sb.append(m.group());
			first = false;
		}
		// autogenerate key name, appended with cleaned origKeyName to be human-friendly
		String newResName = "res_" + resRef;
		String cleanedResName = sb.toString();
		if (!cleanedResName.isEmpty()) {
			newResName += "_" + cleanedResName.toLowerCase();
		}
		return newResName;

Reply:
updated, thanks :)

Reply:
btw, is there any chance for the origKeyName to have non-latin text? (ie. chinese/japanese/korean font)